test/common/json/config_schemas_test.cc

```diff
@@ -0,0 +1,69 @@
+#include "common/json/config_schemas.h"
+#include "common/json/json_loader.h"
+
+#include "test/test_common/environment.h"
+
+#include <dirent.h>
+
+using testing::_;
+
+namespace Json {
+
+std::vector<std::string> generateTestInputs() {
+  TestEnvironment::exec(
+      {TestEnvironment::runfilesPath("test/common/json/test_data/generate_test_data.py")});
+
+  std::string test_path = TestEnvironment::temporaryDirectory();
+  DIR* dir = opendir(test_path.c_str());
+  if (!dir) {
+    throw std::runtime_error("Generated config schema test directory not found");
+  }
+
+  dirent* entry;
+  std::vector<std::string> file_names;
+  while ((entry = readdir(dir)) != nullptr) {
+    std::string file_name = fmt::format("{}/{}", test_path, std::string(entry->d_name));
+    if (entry->d_type == DT_DIR) {
+      continue;
+    }
+    file_names.push_back(file_name);
+  }
+  closedir(dir);
+
+  return file_names;
+}
+
+class ConfigSchemasTest : public ::testing::TestWithParam<std::string> {};
+
+TEST_P(ConfigSchemasTest, CheckValidationExpectation) {
+  ObjectPtr json = Factory::LoadFromFile(GetParam());
+
+  // lookup schema in test input
+  std::string schema, schema_name{json->getString("schema")};
+  if (schema_name == "LISTENER_SCHEMA") {
+    schema = Schema::LISTENER_SCHEMA;
+  } else if (schema_name == "HTTP_CONN_NETWORK_FILTER_SCHEMA") {
+    schema = Schema::HTTP_CONN_NETWORK_FILTER_SCHEMA;
+  } else if (schema_name == "ROUTE_CONFIGURATION_SCHEMA") {
+    schema = Schema::ROUTE_CONFIGURATION_SCHEMA;
+  } else if (schema_name == "ROUTE_ENTRY_CONFIGURATION_SCHEMA") {
+    schema = Schema::ROUTE_ENTRY_CONFIGURATION_SCHEMA;
+  } else if (schema_name == "CLUSTER_SCHEMA") {
+    schema = Schema::CLUSTER_SCHEMA;
+  } else if (schema_name == "TOP_LEVEL_CONFIG_SCHEMA") {
+    schema = Schema::TOP_LEVEL_CONFIG_SCHEMA;
+  } else {
+    FAIL() << fmt::format("Did not recognize schema name {}", schema_name);
+  }
+
+  // perform validation and verify result
+  // TODO(danielhochman): add support for EXPECT_THROW_WITH_MESSAGE
+  if (json->getBoolean("throws", false)) {
+    EXPECT_THROW(json->getObject("data")->validateSchema(schema), Exception);
+  } else {
+    json->getObject("data")->validateSchema(schema);
```

Comment:
I might try/catch here, catch exception, fail, and then print some info about what failed (which test file, etc.), since I think it's going to be almost impossible to figure it out if something starts failing.